Winleafs-Server/Winleafs.Server.Api/Controllers/SpotifyController.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+using System;
+using System.Threading.Tasks;
+using Winleafs.Server.Services.Exceptions;
+using Winleafs.Server.Services.Helpers;
+using Winleafs.Server.Services.Interfaces;
+
+namespace Winleafs.Server.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class SpotifyController : BaseApiController
+    {
+        private IUserService _userService;
+        private ISpotifyService _spotifyService;
+
+        public SpotifyController(DbContext context, IUserService userService, ISpotifyService spotifyService) : base(context)
+        {
+            _userService = userService;
+            _spotifyService = spotifyService;
+        }
+
+        /// <remarks>
+        /// Must be GET due to Spotify authorization standards
+        /// </remarks>
+        [HttpGet]
+        [Route("authorize/{applicationId}")]
+        public async Task<IActionResult> Authorize(string applicationId)
```

Comment:
We should look into rate limiting our API. An API shouldn't call the API multiple times a second unless they are abusing it.
This is an endpoint that could be a weakness and would be good to prevent

Reply:
Future work, whateverr